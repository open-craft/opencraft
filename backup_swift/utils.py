# -*- coding: utf-8 -*-
#
# OpenCraft -- tools to aid developing and hosting free software projects
# Copyright (C) 2015-2019 OpenCraft <contact@opencraft.com>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU Affero General Public License as
# published by the Free Software Foundation, either version 3 of the
# License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Affero General Public License for more details.
#
# You should have received a copy of the GNU Affero General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#
"""
Misc utils for backup_swift.
"""

# Imports #####################################################################

import contextlib
import logging
import requests

# Logging #####################################################################

logger = logging.getLogger(__name__)

# Functions #####################################################################


def ping_heartbeat_url(url):
    """
    Pings heartbeat url.

    :param str url: Url to ping
    :return: True if http request finished and returned proper status, False otherwise.
    """
    try:
        response = requests.get(url, timeout=30)
        return 200 <= response.status_code < 300
    except requests.RequestException:
        return False


@contextlib.contextmanager
def filter_logger(logger_name, new_filter):
    """
    Contextmanager to add a filter to a logger temporarily.
    :param str logger_name: Name of a logger to modify
    :param callable new_filter: Filter to add
    """
    changed_logger = logging.getLogger(logger_name)
    try:
        changed_logger.addFilter(new_filter)
        yield
    finally:
        changed_logger.removeFilter(new_filter)


def filter_swift(log_record):
    """
    Filters log records containing information about the fact that downloaded file was not modified.
    Swift backup spawns quite a log of these, which are not an error condition.
    """
    return '304 Not Modified' not in log_record.getMessage()
