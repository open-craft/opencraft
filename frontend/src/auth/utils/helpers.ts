import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
