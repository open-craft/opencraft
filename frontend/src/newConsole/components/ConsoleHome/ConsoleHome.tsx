import * as React from 'react';
import {
  ConsolePage,
} from 'newConsole/components';
import { InstancesModel } from 'console/models';
import { connect } from 'react-redux';
import { RootState } from 'global/state';
import { ThemePreview } from 'console/components';

interface State {}
interface ActionProps {
  clearErrorMessage: Function;
  updateImages: Function;
}
interface StateProps extends InstancesModel {}
interface Props extends StateProps, ActionProps {}

export class ConsoleHomeComponent extends React.PureComponent<Props, State> {
  public render() {
    return (
      <ConsolePage
        contentLoading={this.props.loading}
        showSideBarEditComponent={false}>
        <ThemePreview/>
      </ConsolePage>
    );
  }
}

export const ConsoleHome = connect<
  StateProps,
  ActionProps,
  {},
  Props,
  RootState
>((state: RootState) => state.console)(ConsoleHomeComponent);
