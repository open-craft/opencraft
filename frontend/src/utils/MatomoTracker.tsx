import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
