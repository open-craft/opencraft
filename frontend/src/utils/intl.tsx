import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
