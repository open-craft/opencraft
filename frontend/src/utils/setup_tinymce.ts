import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
