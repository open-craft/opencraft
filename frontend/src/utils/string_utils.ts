export const toCamelCase = (s: string) => {
  return s.replace(/([-_][a-z])/gi, ($1) => {
    return $1.toUpperCase().replace('-', '').replace('_', '');
  });
};

export const sanitizeErrorFeedback = (input: {
  [key: string]: Array<string>;
}) => {
  // Loop at each error message and join them.
  // Also convert keys from snake_case to camelCase
  const newObject: any = {};

  Object.keys(input).forEach((key) => {
    newObject[toCamelCase(key)] = input[key].join();
  });

  return newObject;
};

export const capitalizeFirstLetter = (str: string) => {
  return str[0].toUpperCase() + str.slice(1);
};
