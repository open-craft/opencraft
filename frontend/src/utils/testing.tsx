import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
