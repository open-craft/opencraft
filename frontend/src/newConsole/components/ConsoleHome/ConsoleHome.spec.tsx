import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { ConsoleHome } from './ConsoleHome';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(<ConsoleHome />).toJSON();
    expect(tree).toMatchSnapshot();
});
