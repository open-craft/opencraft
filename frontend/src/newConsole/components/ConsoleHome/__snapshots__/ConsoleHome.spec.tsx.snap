// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="console-page"
>
  <div
    className="d-flex justify-content-between redeployment-toolbar px-4 py-3 container-fluid"
  >
    <div
      className="d-flex flex-row justify-content-between w-100"
    >
      <span
        className="status-pill badge badge-pill badge-primary"
        style={
          Object {
            "backgroundColor": "grey",
          }
        }
      >
        <div
          className="text"
        >
          Loading
        </div>
        <i
          className="fas fa-question-circle fa-lg"
          onBlur={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        />
      </span>
      <button
        className="float-right btn-disabled btn btn-primary btn-lg"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        <div
          className="button-text-layer"
        >
          Publish changes
        </div>
      </button>
    </div>
    <div
      className="toast-container"
    />
  </div>
  <div
    className="new-console-page-container"
  >
    <div
      className="new-console-page-content row"
    >
      <div
        className="pr-4 pl-0 container-fluid"
      >
        <div
          className="justify-content-center m-0 row"
        >
          <div
            className="p-0 m-0 col-md-3"
          >
            <div
              className="new-customization-menu accordion"
            >
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Theme
                </div>
                <div
                  aria-expanded={null}
                  className="collapse show"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/theming/preview-and-colors"
                        onClick={[Function]}
                      >
                        Colors
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/logos"
                        onClick={[Function]}
                      >
                        Logos
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/buttons"
                        onClick={[Function]}
                      >
                        Buttons
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/navigation"
                        onClick={[Function]}
                      >
                        Navigation
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/footer"
                        onClick={[Function]}
                      >
                        Footer
                      </a>
                      <a
                        aria-current={null}
                        href="/console/theming/hero"
                        onClick={[Function]}
                      >
                        Hero
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Custom Pages
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/custom-pages/about"
                        onClick={[Function]}
                      >
                        About
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/contact"
                        onClick={[Function]}
                      >
                        Contact
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/donate"
                        onClick={[Function]}
                      >
                        Donate
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/tos"
                        onClick={[Function]}
                      >
                        Terms of Service
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/honor"
                        onClick={[Function]}
                      >
                        Honor Code
                      </a>
                      <a
                        aria-current={null}
                        href="/console/custom-pages/privacy"
                        onClick={[Function]}
                      >
                        Privacy Policy
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Instance Settings
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/settings/general"
                        onClick={[Function]}
                      >
                        General
                      </a>
                      <a
                        aria-current={null}
                        href="/console/settings/domain"
                        onClick={[Function]}
                      >
                        Domain
                      </a>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card"
              >
                <div
                  className="card-header"
                  onClick={[Function]}
                  variant="link"
                >
                  Courses (Studio)
                </div>
                <div
                  aria-expanded={null}
                  className="collapse"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="flex-column nav card-header-undefined"
                      onKeyDown={[Function]}
                    >
                      <a
                        aria-current={null}
                        href="/console/courses/manage"
                        onClick={[Function]}
                      >
                        Manage
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="pr-0 col-md-9"
          >
            <div
              className="customization-form"
            >
              <div
                className="loading"
              >
                <i
                  className="fas fa-2x fa-sync-alt fa-spin"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
